realm/realm-library/src/androidTest/kotlin/io/realm/NullMixedSetTester.kt

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import io.realm.entities.AllTypes
+import io.realm.rule.BlockingLooperThread
+import java.util.*
+import kotlin.reflect.KFunction1
+import kotlin.test.*
+
+/**
+ * Tester for Mixed containing null values. It evaluates that null is accepted as a value and that only a null instance
+ * can exist in a RealmSet.
+ */
+class NullMixedSetTester(
```

Comment:
Might be that I lost the overview on this, but couldn't all the null checks just have been tested implicit by adding some null values to the data sets of the`ManagedSetTester<Mixed>` case? Seems a bit overkill with a complete tester for this and then we also don't test mixing nulls with values.

Reply:
It is a special case because we test with two valid null mixed values: `null` and `Mixed.nullValue`. We could pass the expected set size, in case that there are duplicates, but it would also add more complexity to the test suite.